src/qml/geometry_editors/SplitFeatureToolbar.qml

```diff
@@ -23,13 +23,23 @@ VisibilityFadingRow {
     onConfirm: {
       // TODO: featureModel.currentLayer.selectByIds([featureModel.feature.id], VectorLayerStatic.SetSelection)
       Utils.selectFeaturesInLayer(featureModel.currentLayer, [featureModel.feature.id], VectorLayerStatic.SetSelection)
-      var line = drawLineToolbar.rubberbandModel.pointSequence(featureModel.currentLayer.crs)
       if (!featureModel.currentLayer.editBuffer())
         featureModel.currentLayer.startEditing()
-      featureModel.currentLayer.splitFeatures(line, true)
-      featureModel.currentLayer.commitChanges()
-      cancel()
-      finished()
+
+      var result = GeometryUtils.splitFeatureFromRubberBand(featureModel.currentLayer, drawLineToolbar.rubberbandModel)
+      if ( result !== QgsGeometryStatic.Success )
+      {
+        displayToast( qsTr( 'Feature could not be split' ) );
+        featureModel.currentLayer.rollBack()
+        cancel()
+        finished()
+      }
+      else
+      {
+        featureModel.currentLayer.commitChanges()
+        cancel()
```

Comment:
It seems weird to "cancel" after a successful operation.
Should this be called "closeToolbar". And ideally be directly connected to finished()?